### PR TITLE
MAINT: Fix bugs with code credit and linking

### LINKED ISSUE #12817: bug in auto-linking of PRs in changelog
Links to PRs in the changelog (https://mne.tools/stable/changes/v1.8.html) don't work. They link to `https://github.com/mne-tools/mne-python/pulls/00000` but it should be "pull" not "pulls"

### LINKED ISSUE #12818: code credit dedup
Some possible deduplications:

- `Jean-Remi KING` and `Jean-Remi King` are separate entries
- `Valerii` and `Valerii Chirkov`
- `PAB` and `Pierre-Antoine Bannier`
- `Jordan Drew` and `Drew, J.`
- `Deleted user` still shows up (even though it's trying to be excluded in `update_credit_rst.py`)

Something I think we discussed before, but I can't recall what was decided:

- there are some folks who I'm pretty sure are in our `.mailmap` but their usernames aren't getting parsed out here (`TzionaN`, `Sena`, and `Martin` are a few examples I noticed; probably there are more). Did we decide not to use `.mailmap` for some reason? Or is it a bug that these users aren't getting the preferred name displayed?

### PR DESCRIPTION
**Pull request by larsoner**
Closes #12817
Closes #12818

Also fix the neo part of https://app.circleci.com/pipelines/github/mne-tools/mne-python/25126/workflows/44f5e760-f46a-4e48-8a46-775135dcee05/jobs/68338, need to wait for Dipy to push out new wheels for other fixes.

### CODE REVIEW

File: doc/sphinxext/credit_tools.py [hunk 1]
```diff
@@ -26,26 +26,68 @@
...
+    "alexandra": "Alexandra Corneyllie",  # 7600
+    "alexandra.corneyllie": "Alexandra Corneyllie",  # 7600
+    "akshay0724": "Akshay",  # 4046, TODO: Check singleton
+    "AnneSo": "Anne-Sophie Dubarry",  # 4910
+    "Basile": "Basile Pinsard",  # 1791
+    "ChristinaZhao": "Christina Zhao",  # 9075
+    "Drew, J.": "Jordan Drew",  # 10861
+    "Hamid": "Hamid Maymandi",  # 10849
+    "jwelzel": "Julius Welzel",  # 11118
+    "Jean-Rémi King": "Jean-Rémi King",
```
Comment (drammock) [inline doc/sphinxext/credit_tools.py:55]: this one looks vacuous, maybe a "find/replace-all" mistake?

File: doc/sphinxext/credit_tools.py [hunk 2]
```diff
@@ -84,6 +126,18 @@ def generate_credit_rst(app=None, *, verbose=False):
 
     unknown_emails: set[str] = set()
 
+    # Use doc/changes/*.rst to map PR numbers to names
```
Comment (drammock) [inline doc/sphinxext/credit_tools.py:129]: a bit concerned about this, as (1) authors listed in changelog won't necessarily match those in `Co-authored-by` commit trailers, and (2) for older changelog entries, we must take in on faith that the PR number was entered correctly.

File: doc/sphinxext/mne_doc_utils.py [hunk 3]
```diff
@@ -159,6 +161,11 @@ def reset_warnings(gallery_conf, fname):
         "A worker stopped while some jobs were given to the executor.*",
         category=UserWarning,
     )
+    # neo
+    warnings.filterwarnings(
+        "ignore",
+        "The 'copy' argument in Quantity is deprecated.*",
+    )
```
Comment (drammock) [inline doc/sphinxext/mne_doc_utils.py:168]: need equivalent in conftest too:

https://github.com/mne-tools/mne-python/actions/runs/10617877023/job/29431850995?pr=12820#step:17:5278

**Review by agramfort — APPROVED** [review]

**Comment by drammock** [comment]
hmm, looks like the `quantities.QuantitiesDeprecationWarning` only exists for the last 2 months. So just adding it to `conftest.py` will only work for CIs that get quantities 0.16.0. Ones that aren't getting that are old, minimal, and pip-pre

File: doc/sphinxext/credit_tools.py (hunk 1 above)
Comment (larsoner) [reply doc/sphinxext/credit_tools.py]: Yep, removed!

**Pull request merged**